### ISSUE TITLE
Adding a new contact sets the displayname to the current displayname, instead of leaving it empty so it can change when the contact changes their name

### ISSUE THREAD

**Issue by missytake** [deltachat/deltachat-desktop#4045]
- Operating System (Linux/Mac/Windows/iOS/Android): xubuntu 22.04
- Delta Chat Version: 1.46.0
- Expected behavior: For contacts where I never set a display name manually, the display name changes when the contact changes it.
- Actual behavior: The display name they had when I met them stays, even if they change it, unless I explicitly remove it in the UI.
- Steps to reproduce the problem:
  - click on an invite link to add a contact, e.g. `OPENPGP4FPR:AA5FDEF02BFC355FDEA09FF4CA4AFCD2F065E613#a=deltawoot%40nine.testrun.org&n=Friendly%20Chatwoot&i=q4DhTVr1T2A&s=mT3Bo9JDdVx`
  - click on their name to display contact information
  - click edit in the top right to edit the display name
  - realize that it is not empty, but set to the current name
  - exit without changes
  - from the other profile, change the display name 
  - from the other profile, send a message to yourself
  - realize the display name in the app is not changed
  - edit the display name to remove it
  - realize that now the display name is the changed one
- Screenshots:

![image](https://github.com/user-attachments/assets/3a2948cd-2fae-4d87-8f81-be5f705a4230)

**Comment by Simon-Laux** [comment]
I noticed this (names not empty) too, but I didn't know it was related to invite links

**Comment by nicodh** [comment]
I think this is solved meanwhile. I can't reproduce it and I added some e2e tests in #5159 which work fine too

**Comment by WofWca** [comment]
Yes, it looks like this is solved, or maybe needs more accurate repro steps.

I generated an invite link (i.delta.chat), then opened it in the browser, then pressed "open chat". This opened Delta Chat with a dialog "Chat with Bob?". I pressed "yes", then opened the contact info, then "Edit name", and it was empty.

**Comment by WofWca** [comment]
Additionally, I checked the precise link from the issue description, and the contact name updated immediately after E2E established.
